src/Config.h

```diff
@@ -0,0 +1,1488 @@
+        }
+
+        void setSteamLedInverted(const bool value) {
+            _doc["hardware"]["leds"]["steam"]["inverted"] = value;
+        }
+
+        // Hardware - Sensors
+        bool getPressureSensorEnabled() {
+            return _doc["hardware"]["sensors"]["pressure"]["enabled"] | false;
+        }
+
+        void setPressureSensorEnabled(const bool value) {
+            _doc["hardware"]["sensors"]["pressure"]["enabled"] = value;
+        }
+
+        bool getWaterTankSensorEnabled() {
+            return _doc["hardware"]["sensors"]["watertank"]["enabled"] | false;
+        }
+
+        void setWaterTankSensorEnabled(const bool value) {
+            _doc["hardware"]["sensors"]["watertank"]["enabled"] = value;
+        }
+
+        int getWaterTankSensorMode() {
+            return _doc["hardware"]["sensors"]["watertank"]["mode"];
+        }
+
+        void setWaterTankSensorMode(const int value) {
+            _doc["hardware"]["sensors"]["watertank"]["mode"] = value;
+        }
+
+        // Scale
+        bool getScaleEnabled() {
+            return _doc["hardware"]["sensors"]["scale"]["enabled"] | false;
+        }
+
+        void setScaleEnabled(const bool value) {
+            _doc["hardware"]["sensors"]["scale"]["enabled"] = value;
+        }
+
+        int getScaleSamples() {
+            return _doc["hardware"]["sensors"]["scale"]["samples"];
+        }
+
+        void setScaleSamples(const int value) {
+            _doc["hardware"]["sensors"]["scale"]["samples"] = value;
+        }
+
+        int getScaleType() {
+            return _doc["hardware"]["sensors"]["scale"]["type"];
+        }
+
+        void setScaleType(const int value) {
+            _doc["hardware"]["sensors"]["scale"]["type"] = value;
+        }
+
+        float getScaleCalibration() {
+            return _doc["hardware"]["sensors"]["scale"]["calibration"] | static_cast<float>(SCALE_CALIBRATION_FACTOR);
+        }
+
+        void setScaleCalibration(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["calibration"] = constrain(value, -100000.0f, 100000.0f);
+        }
+
+        void setScaleKnownWeight(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["known_weight"] = constrain(value, 0.0f, 2000.0f);
+        }
+
+        float getScaleKnownWeight() {
+            return _doc["hardware"]["sensors"]["scale"]["known_weight"] | static_cast<float>(SCALE_KNOWN_WEIGHT);
+        }
+
+        float getScale2Calibration() {
+            return _doc["hardware"]["sensors"]["scale"]["calibration2"] | static_cast<float>(SCALE_CALIBRATION_FACTOR);
+        }
+
+        void setScale2Calibration(const float value) {
+            _doc["hardware"]["sensors"]["scale"]["calibration2"] = constrain(value, -100000.0f, 100000.0f);
+        }
+
+    private:
+        inline static auto CONFIG_FILE = "/config.json";
+
+        StaticJsonDocument<4096> _doc;
+
+        bool validateAndApplyConfig(const JsonDocument& doc) {
```

Comment:
This file is *really* long and for the sake of legibility I would separate header and implementation. That makes it a lot easier to understand the structure of the class.